internal/gcp/secret-manager.go

```diff
@@ -0,0 +1,176 @@
+	if s.client == nil {
+		s.logf("gcp: no connection to GCP secret manager: '%s'", s.ProjectID)
+		return errCreateKey
+	}
+
+	existingSecret, _ := s.client.GetSecret(
+		context.Background(),
+		&secretmanagerpb.GetSecretRequest{
+			Name: s.getSecretName(s.ProjectID, key),
+		},
+	)
+
+	if existingSecret != nil {
+		return kes.ErrKeyExists
+	}
+
+	secret, err := s.client.CreateSecret(
+		context.Background(),
+		&secretmanagerpb.CreateSecretRequest{
+			Parent:   fmt.Sprintf("projects/%s", s.ProjectID),
+			SecretId: key,
+			Secret: &secretmanagerpb.Secret{
+				Replication: &secretmanagerpb.Replication{
+					Replication: &secretmanagerpb.Replication_Automatic_{
+						Automatic: &secretmanagerpb.Replication_Automatic{},
+					},
+				},
+			},
+		},
+	)
+
+	if err != nil {
+		s.logf("gcp: failed to create '%s': %v", key, err)
+		return errCreateKey
+	}
+
+	_, err = s.client.AddSecretVersion(
+		context.Background(),
+		&secretmanagerpb.AddSecretVersionRequest{
+			Parent: secret.Name,
+			Payload: &secretmanagerpb.SecretPayload{
+				Data: []byte(value),
+			},
+		},
+	)
+
+	if err != nil {
+		s.logf("gcp: failed to create '%s': %v", key, err)
+		return errCreateKey
+	}
+
+	return nil
+}
+
+// Get returns the value associated with the given key.
+func (s *SecretManager) Get(key string) (string, error) {
+	if s.client == nil {
+		s.logf("gcp: no connection to GCP secret manager: '%s'", s.ProjectID)
+		return "", errGetKey
+	}
+
+	result, err := s.client.AccessSecretVersion(
+		context.Background(),
+		&secretmanagerpb.AccessSecretVersionRequest{
+			Name: s.getSecretName(s.ProjectID, key),
+		},
+	)
+
+	if err != nil {
+		s.logf("gcp: failed to read '%s': %v", key, err)
+		return "", errGetKey
+	}
+
+	secret := string(result.Payload.Data)
+	return secret, nil
+}
+
+// Delete remove key-value pair from GCP secret manager
+func (s *SecretManager) Delete(key string) error {
+	if s.client == nil {
+		s.logf("gcp: no connection to GCP secret manager: '%s'", s.ProjectID)
+		return errDeleteKey
+	}
+
+	err := s.client.DeleteSecret(
+		context.Background(),
+		&secretmanagerpb.DeleteSecretRequest{
+			Name: s.getSecretName(s.ProjectID, key),
+		},
+	)
+
+	if err != nil {
+		s.logf("gcp: failed to delete '%s': %v", key, err)
+		return errDeleteKey
+	}
+
+	return nil
+}
+
+// Authenticate tries to auth and connect to GCP secret manager
+// using the given credentials file
+func (s *SecretManager) Authenticate() error {
+	client, err := secretmanager.NewClient(
+		context.Background(),
+		option.WithCredentialsFile(s.CredentialsFile),
+	)
+	if err != nil {
+		return err
+	}
+	s.client = client
+	return nil
+}
+
+func (s *SecretManager) logf(format string, v ...interface{}) {
+	if s.ErrorLog == nil {
+		log.Printf(format, v...)
+	} else {
+		s.ErrorLog.Printf(format, v...)
+	}
+}
+
+func (s *SecretManager) getSecretName(projectID, secretID string) string {
+	return fmt.Sprintf("projects/%s/secrets/%s/versions/1", projectID, secretID)
```

Comment:
```suggestion
	return path.Join("projects", projectID, "secrets", secretID, "versions/1")
```
Handles path segments properly...